src/Lean/Meta/IndPredBelow.lean

```diff
@@ -227,6 +230,24 @@ def mkBelowDecl (ctx : Context) : MetaM Declaration := do
     (←ctx.typeInfos.mapIdxM $ mkInductiveType ctx).toList
     ctx.typeInfos[0].isUnsafe
 
+partial def backwardsChaining (m : MVarId) (depth : Nat) : MetaM Bool := do
+  if depth = 0 then false
+  else
+    withMVarContext m do
+    let lctx ← getLCtx 
+    let mTy ← getMVarType m
+    lctx.anyM fun localDecl => 
+      if localDecl.isAuxDecl then false else
+      commitWhen do
+      let (mvars, _, t) ← forallMetaTelescope localDecl.type
+      if ←isDefEq mTy t then 
+        assignExprMVar m (mkAppN localDecl.toExpr mvars)
+        mvars.allM fun v => do
+          if ←isExprMVarAssigned v.mvarId! then true
```

Comment:
We have use `<||>` the monadic or.
https://github.com/leanprover/lean4/blob/master/src/Init/Control/Basic.lean#L49-L64